tests/upgrade_integration/upgrade_test.py

```diff
@@ -0,0 +1,61 @@
+import subprocess
+import shutil
+from time import sleep
+
+from unittest import TestCase
+
+from dusty.commands.upgrade import _test_dusty_binary
+from dusty import constants
+
+from ..testcases import DustyIntegrationTestCase
+
+class TestUpgrade(DustyIntegrationTestCase):
+    def tearDown(self):
+        super(TestUpgrade, self).tearDown()
+        self.stop_daemon()
+
+    def run_daemon_binary(self, path='./dist/dusty'):
+        self.daemon_process = subprocess.Popen(args=[path, '-d', '--suppress-warnings'], stdout=subprocess.PIPE)
+        sleep(1)
+
+    def run_daemon_source(self):
+        self.daemon_process = subprocess.Popen(args=['dusty', '-d', '--suppress-warnings'], stdout=subprocess.PIPE)
+        sleep(1)
+
+    def stop_daemon(self):
+        self.daemon_process.terminate()
+
+    def recreate_dusty_binary(self):
+        subprocess.check_call(['./setup/create_binaries.sh'], stdout=subprocess.PIPE)
+
+    def test_upgrade_2_1(self):
+        self.run_daemon_binary()
+        version = '0.2.1'
+        output = self.run_command('version')
+        self.assertInSameLine(output, 'daemon', 'version')
+        output = self.run_command('upgrade {}'.format(version))
+        self.assertInSameLine(output, 'Downloading', version)
+        self.assertInSameLine(output, 'Finished upgrade', version)
+        sleep(1)
+        output = self.run_command('version', raise_on_error=False)
```

Comment:
Can't you just catch this instead of adding this `raise_on_error` flag you only use for testing?